CMAKE/CheckLAPACKCompilerFlags.cmake

```diff
@@ -14,6 +14,19 @@ macro( CheckLAPACKCompilerFlags )
 
 set( FPE_EXIT FALSE )
 
+# FORTRAN ILP default
+if ( FORTRAN_ILP )
+    if( CMAKE_Fortran_COMPILER_ID STREQUAL "Intel" )
+        if ( WIN32 )
+            set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} /integer-size:64")
+        else ()
+            set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -integer-size 64")
+        endif()
+    else()
+        set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fdefault-integer-8")
```

Comment:
@weslleyspereira The [logic](https://github.com/Reference-LAPACK/lapack/pull/617/files) in #617 needs to go here.